drivers/linstor-manager

```diff
@@ -779,6 +779,119 @@ def destroy_drbd_volumes(session, args):
     return str(False)
 
 
+def health_check(session, args):
+    group_name = args['groupName']
+
+    result = {
+       'controller-uri': '',
+       'nodes': {},
+       'storage-pools': {},
+       'warns': [],
+       'errors': []
+    }
+
+    def format_result():
+        return json.dumps(result)
+
+    # 1. Get controller.
+    try:
+        controller_uri = get_controller_uri()
+
+        result['controller-uri'] = controller_uri
+        try:
+            if controller_uri == 'linstor://localhost':
+                # Replace `localhost` with IP to give a better info for users.
+                result['controller-uri'] = 'linstor://' + util.get_this_host_address(session)
+        except Exception:
+            # Ignore error: can be a XAPI restart or something else.
+            pass
+
+        linstor = LinstorVolumeManager(
+            controller_uri,
+            group_name,
+            logger=util.SMlog
+        )
+    except Exception as e:
+        # Probably a network issue, or offline controller.
+        result['errors'].append('Cannot join SR: `{}`.'.format(e))
+        return format_result()
+
+    try:
+        # 2. Check node statuses.
+        nodes = linstor.get_nodes_info()
+        result['nodes'] = nodes
+        for node_name, status in nodes.items():
+            if status != 'ONLINE':
+                result['warns'].append('Node `{}` is {}.'.format(node_name, status))
+
+        # 3. Check storage pool statuses.
+        storage_pools = linstor.get_storage_pools_info()
+        result['storage-pools'] = storage_pools
+        for node_name, storage_pools in storage_pools.items():
+            for storage_pool in storage_pools:
+                free_size = storage_pool['free-size']
+                capacity = storage_pool['capacity']
+                if free_size < 0 or capacity <= 0:
```

Comment:
capacity can't be 0 here as you already pre-processed it. Maybe it should be left as 0 in the output of get_storage_pools_info?

Reply:
`free_size` has already been pre-processed too, I prefer to have a normalized value and the same for each size attribute in case of error.

Reply:
`free_size` is not an issue as you specifically test for `< 0` which is consistent with the fact that you set it to -1 if the value is not pertinent. To me it's the `<=` operator that is misleading as there's no way the value could be 0. Not a blocker for me.